plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetastoreClosure.java

```diff
@@ -87,13 +88,35 @@ public Set<HiveColumnStatisticType> getSupportedColumnStatistics(Type type)
 
     public PartitionStatistics getTableStatistics(String databaseName, String tableName)
     {
-        return delegate.getTableStatistics(getExistingTable(databaseName, tableName));
+        return getTableStatistics(databaseName, tableName, Optional.empty());
+    }
+
+    public PartitionStatistics getTableStatistics(String databaseName, String tableName, Optional<Set<String>> columns)
+    {
+        Table table = getExistingTable(databaseName, tableName);
+        if (columns.isPresent()) {
+            Set<String> requestedColumnNames = columns.get();
+            List<Column> requestedColumns = table.getDataColumns().stream().filter(column -> requestedColumnNames.contains(column.getName())).collect(toImmutableList());
+            table = Table.builder(table).setDataColumns(requestedColumns).build();
+        }
+        return delegate.getTableStatistics(table);
     }
 
     public Map<String, PartitionStatistics> getPartitionStatistics(String databaseName, String tableName, Set<String> partitionNames)
+    {
+        return getPartitionStatistics(databaseName, tableName, Optional.empty(), partitionNames);
+    }
+
+    public Map<String, PartitionStatistics> getPartitionStatistics(String databaseName, String tableName, Optional<Set<String>> columns, Set<String> partitionNames)
     {
         Table table = getExistingTable(databaseName, tableName);
         List<Partition> partitions = getExistingPartitionsByNames(table, ImmutableList.copyOf(partitionNames));
+        if (columns.isPresent()) {
+            Set<String> requestedColumnNames = columns.get();
+            List<Column> requestedColumns = table.getDataColumns().stream().filter(column -> requestedColumnNames.contains(column.getName())).collect(toImmutableList());
+            table = Table.builder(table).setDataColumns(requestedColumns).build();
```

Comment:
nit. Maybe apply different formatting? Also, this code is same as L96-99
```suggestion
            Set<String> requestedColumnNames = columns.get();
            List<Column> requestedColumns = table.getDataColumns().stream()
                    .filter(column -> requestedColumnNames.contains(column.getName()))
                    .collect(toImmutableList());
```

We could create some method like getProjectedTable:
```java
private static Table getProjectedTable(Table table, Optional<Set<String>> columns)
    {
        if (columns.isPresent()) {
            Set<String> requestedColumnNames = columns.get();
            List<Column> requestedColumns = table.getDataColumns().stream()
                    .filter(column -> requestedColumnNames.contains(column.getName()))
                    .collect(toImmutableList());
            table = Table.builder(table).setDataColumns(requestedColumns).build();
        }
        return table;
    }
``` 
and use it in both places?

Reply:
> We could create some method like getProjectedTable

In this case we also "project" partitions inside the if and need the `requestedColumns` for this.

formatting applied